File: parse_check.js
const fs = require('fs');
const acorn = require('acorn');
const jsx = require('acorn-jsx');

const code = fs.readFileSync('frontend/app/(main)/luu-tru/page.tsx', 'utf8');

try {
  acorn.Parser.extend(jsx()).parse(code, { sourceType: 'module', ecmaVersion: 2020 });
  console.log("No syntax errors found by acorn-jsx");
} catch (e) {
  console.error("Syntax error:", e.message);
}

File: parse_check.py
import re

with open('frontend/app/(main)/luu-tru/page.tsx', 'r') as f:
    code = f.read()

# Very basic check for unmatched braces
braces = 0
for i, c in enumerate(code):
    if c == '{': braces += 1
    elif c == '}': braces -= 1
    
print("Unmatched braces:", braces)
